docs/theory/diffusion/diffusion.rst

```diff
@@ -0,0 +1,667 @@
+In addition to that, MSDs for different lags are highly correlated.
+One should not fit these values as though they were independent data points unless care is taken that it is valid to do so (more on this below).
+
+Ordinary Least Squares
+^^^^^^^^^^^^^^^^^^^^^^
+
+Real measurements are affected by noise, which leads to localization error.
+In addition to this, depending on the method used to detect the position of the particle, there may also be motion blur.
+These two sources of error manifest themselves as an offset in the MSD curve given by:
+
+.. math::
+
+    offset = 2 \sigma^2 - 4 R D \Delta t
+
+Here :math:`\sigma` is the static localization accuracy, :math:`R` is a motion blur constant and :math:`\Delta t` represents the time step.
+With pure diffusive motion (a complete absence of drift) in an isotropic medium, 1-dimensional MSDs can be fitted by the following relation:
+
+.. math::
+
+    \rho[n] = 2 D n \Delta t + offset
+
+Here :math:`D` is the diffusion constant in :math:`um^2/s`, :math:`\Delta t` is the time step, :math:`n` is the step index and the offset is determined by the localization accuracy.
+
+While it may be tempting to use a large number of lags in the diffusion estimation procedure, this actually produces poor estimates of the diffusion constant :cite:`qian1991single,michalet2010mean,michalet2012optimal`.
+There exists an optimal number of lags to fit such that the estimation error is minimal.
+This optimal number of lags depends on the ratio between the diffusion constant and the dynamic localization accuracy:
+
+.. math::
+
+    \epsilon_{localization} = \frac{offset}{slope} = \frac{2 \sigma^2 - 4 R D \Delta t}{2 D \Delta t} = \frac{\sigma^2}{D \Delta t} - 2 R
+
+When localization is infinitely accurate, the optimal number of points is two :cite:`michalet2010mean`.
+At the optimal number of lags, it doesn't matter whether we use a weighted or unweighted least squares algorithm to fit the curve :cite:`michalet2010mean`, and therefore we opt for the latter, analogously to :cite:`michalet2012optimal`.
+With Pylake, you can obtain an estimated diffusion constant by invoking::
+
+    >>> tracks[0].estimate_diffusion(method="ols")
+    DiffusionEstimate(value=7.804440367653842, std_err=2.527045387449447, num_lags=2, num_points=80, method='ols', unit='um^2 / s')
+
+Note that Pylake gives you both an estimate for the diffusion constant, as well as its expected uncertainty and the number of lags used in the computation.
+The uncertainty estimate in this case is based on Equation A1b in :cite:`bullerjahn2020optimal`.
+
+Generalized Least Squares
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Another option is to use generalized least squares (GLS).
+Based on the definition of the MSD estimator, it is possible to derive the structure of the covariance matrix :cite:`bullerjahn2020optimal`.
+This covariance matrix can then be used in the estimation procedure to determine the diffusion constant.
+
```

Comment:
How does this method compare to the one above? Does the method from michalet account for correlations? and do I understand correctly that the GLS method should be combined with a selection of the optimal number of lags?

Reply:
And for OLS you mention the blur constant and localization accuracy, but you don't mention this for the other methods.

Reply:
> And for OLS you mention the blur constant and localization accuracy, but you don't mention this for the other methods.

For `GLS` things are the same as `OLS` and we don't support supplying the motion blur manually. It all gets subsumed in that offset. `GLS` is identical to `OLS` in every way except for the fact that we take into account the covariance matrix and don't have a clear procedure for estimating the number of lags to include.

But good point, for `CVE`! I should mention it specifically. I have added the following:

Here $D$ represents the diffusion constant, $\sigma$ is the localization uncertainty standard deviation and $R$ is the motion blur constant. In the current implementation, we assume the motion blur to be negligible for confocal scans (see note below).

In the note (to try and not distract from the main text too much), I show the reasoning for having a motion blur of zero.

```
    The motion blur coefficient :math:`R` is a value between `0` and `1/4` given by

    .. math::

        R = \frac{1}{\Delta t} \int_{0}^{\Delta t}S(t) \left(1 - S(t)\right)dt

    with

    .. math::

        S(t) = \int_{0}^{t} c(t') dt'

    Here, the aperture function is defined as :math:`c(t)`, where :math:`c(t)` represents the fraction of illumination happening over time.
    :math:`c(t)` is normalized w.r.t. area under the curve.
    For a rectangular shutter or exposure window, one obtains :math:`R = 1/6 \frac{t_{exposure}}{line\_time}`.
    In the current implementation we assume the motion blur constant to be negligible (zero) for confocal acquisition, since the time spent over the particle's PSF compared to the line time is very low.

    When estimating both localization uncertainty and the diffusion constant, the motion blur factor has no effect on the estimate of the diffusion constant itself, but it does affect the calculated uncertainties.
    In the case of a provided localization uncertainty, it does impact the estimate of the diffusion constant itself.
```

I have considered providing a ballpark estimate for the motion blur coefficient based on the pixel size, shape of the scan volume and line time; but I feel that should be its own task (since then one might also want to consider the shape of the PSF). All in all, it requires some additional investigation.

We could consider exposing `R` as an optional parameter for the `CVE` method with a default of zero for now, but that would make things harder once we get camera based kymographs that require a higher `R`. My hope is that one day we can just expose an `particle exposure time` on the `Kymo` that will hook straight into this `R` value here.

For camera-based tracking R becomes much more relevant as it begins to approach `1/6`.

Reply:
Thanks for adding the note and explanation! If the motion blur is also negligible for the ensemble approach then I guess it's fine to not expose R for now. And to wait for the camera based kymos.

Reply:
For typical Kymos that I've seen in house, the effect is very small (the spot size is small compared to the width of the Kymo). I think for the purpose of this document, for now, we just explicitly state that for `CVE` we have it set to be zero.